lib/netext/dialer.go

```diff
@@ -66,6 +82,40 @@ func (b BlackListedIPError) Error() string {
 	return fmt.Sprintf("IP (%s) is in a blacklisted range (%s)", b.ip, b.net)
 }
 
+// init detects available IP network versions.
+func init() {
+	detectInterfaces()
+}
+
+// detectInterfaces detects available IP network versions.
+func detectInterfaces() {
```

Comment:
I think this should be moved to some initialization of each `Dialer` , and put the ip6,ip4 globalvars as fields of the Dialer. You could use sync.Once if no good place for initialization exists, although I doubt this